File: Programmers/[77484] 로또의 최고 순위와 최저 순위/ONE/77484.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int containCount = 0;
        int zeroCount = 0;
        Map<Integer, Integer> map = new HashMap<>();
        Set<Integer> win_num = new HashSet<>();

        for (int num : win_nums)
            win_num.add(num);

        for(int i = 1; i <= 6; i++)
            map.put(7 - i, i);
        map.put(0, 6);

        for (int num : lottos) {
            if (num == 0) {
                zeroCount++;
                continue;
            }
            if(win_num.contains(num))
                containCount++;
        }
        answer[0] = map.get(containCount + zeroCount);
        answer[1] = map.get(containCount);

        return answer;
    }
}

File: Programmers/[77484] 로또의 최고 순위와 최저 순위/JFe/77484.py
def solution(lottos, win_nums):
    correct, zero, answer = 0, 0, []
    for l in lottos:
        # 0 체크
        if l == 0:
            zero += 1
            continue
        # 맞는 번호 체크
        for n in win_nums:
            if l == n: correct += 1
    # 최고 순위
    if correct + zero > 1: answer.append(7 - correct - zero)
    else: answer.append(6)
    # 최저 순위
    if correct > 1: answer.append(7 - correct)
    else: answer.append(6)
    return answer